explorer/explorer.go

```diff
@@ -135,9 +140,50 @@ func (exp *explorerUI) rootWebsocket(w http.ResponseWriter, r *http.Request) {
 				exp.wsHub.HubRelay <- sigPingAndUserCount
 			}
 		}()
-
+		go func() {
+			for {
+				msg := &WebSocketMessage{}
+				websocket.JSON.Receive(ws, &msg)
+				var webData WebSocketMessage
+				switch msg.EventId {
+				case "decodetx":
+					webData.EventId = "decodedtx"
+					log.Debug("Recieved decodetx signal for hex: ", msg.Messsage)
+					tx := exp.blockData.DecodeRawTransaction(msg.Messsage)
+					if tx != nil {
+						buff := new(bytes.Buffer)
+						enc := json.NewEncoder(buff)
+						enc.SetIndent("", "    ")
+						enc.Encode(tx)
+						webData.Messsage = buff.String()
+					} else {
+						log.Debugf("Could not decode raw tx")
+						webData.Messsage = fmt.Sprintf("Error: Could not decode hex %s", msg.Messsage)
+					}
+				case "sendtx":
+					webData.EventId = "senttx"
+					log.Debug("Recieved sendtx signal for hex: ", msg.Messsage)
+					txid, err := exp.blockData.SendRawTransaction(msg.Messsage)
```

Comment:
a final check on message length here is warranted IMO

Reply:
How long can the message be?

Reply:
Looking at https://godoc.org/golang.org/x/net/websocket#Codec.Receive, it seems the default max it can accept is 32MB, is that too much?

Reply:
Good find.  Since the block size is 1 MB, I think 32 MB for a tx is too big.  For whatever reason, we can just set this at 2 MB.

Also, we'll need to check the error code and deal with the next read containing the remaining data:

> The whole frame payload is read to an in-memory buffer; max size of payload is defined by `ws.MaxPayloadBytes`. If frame payload size exceeds limit, `ErrFrameTooLarge` is returned; in this case frame is not read off wire completely. The next call to Receive would read and discard leftover data of previous oversized frame before processing next frame. ```

Reply:
I think what it says here ```The next call to Receive would read and discard leftover data of previous oversized frame before processing next frame. ``` means that we don't need to deal with the extra data ourselves as Receive would discard the leftover data before processing the next frame, but I could be wrong.

Reply:
Yeah, I think you're right.  The key is `before processing next frame`.

Reply:
Oh, I guess we can't change the 32MB value.  But if someone submits a tx larger than 1MB, it's definitely not OK to broadcast.